search/api.py

```diff
@@ -90,5 +90,8 @@ def execute_search(*, user, query):
     search = Search(index=index, using=get_conn())
     search.update_from_dict(query)
     channel_names = [channel.display_name for channel in Api(user).list_channels()]
-    search = search.filter("terms", channel_name=channel_names)
+    channels_filter = Q("terms", author_channel_membership=channel_names) | Q(
```

Comment:
Is dealing with public channels going to be in another PR?  This line will prevent searching in any channel that the user is not subscribed to (ie public channels, or conceivably a private channel which the user is a contributor/moderator on but not a subscriber?).

Reply:
I'm going to change this query to restrict to only the channel being viewed, or no channels if the user does not have permission to view that channel.

Reply:
Actually I changed it to use the models for contributor and moderator membership. It would be a significant amount of work to filter only on the user and since this is still behind a feature flag I think we can have a PR to add the public channel part of it before it is open to the public